packages/contracts/src/StakeManager.sol

```diff
@@ -179,4 +212,42 @@ contract StakeManager is IStakeManager, Ownable {
         stakes[relayManager].token.safeTransfer(beneficiary, reward);
         emit StakePenalized(relayManager, beneficiary, stakes[relayManager].token, reward);
     }
+
+    /// @inheritdoc IStakeManager
+    function isRelayEscheatable(address relayManager) public view override returns (bool) {
+        IStakeManager.StakeInfo memory stakeInfo = stakes[relayManager];
+        return stakeInfo.abandonedTime != 0 && stakeInfo.abandonedTime + escheatmentDelay < block.timestamp;
+    }
+
+    /// @inheritdoc IStakeManager
+    function markRelayAbandoned(address relayManager) external override onlyOwner {
+        StakeInfo storage info = stakes[relayManager];
+        require(info.stake > 0, "relay manager not staked");
+        require(info.abandonedTime == 0, "relay manager already abandoned");
+        require(info.keepaliveTime + abandonmentDelay < block.timestamp, "relay manager was alive recently");
+        info.abandonedTime = block.timestamp;
+        emit RelayAbandoned(relayManager, info.abandonedTime);
+    }
+
+    /// @inheritdoc IStakeManager
+    function escheatAbandonedRelayStake(address relayManager) external override onlyOwner {
+        StakeInfo storage info = stakes[relayManager];
+        require(isRelayEscheatable(relayManager), "relay server not escheatable yet");
+        uint256 amount = info.stake;
+        info.stake = 0;
+        info.withdrawTime = 0;
+        info.token.safeTransfer(devAddress, amount);
+        emit AbandonedRelayManagerStakeEscheated(relayManager, msg.sender, info.token, amount);
+    }
+
+    /// @inheritdoc IStakeManager
+    function updateRelayKeepaliveTime(address relayManager) external override {
+        StakeInfo storage info = stakes[relayManager];
+        bool isHubAuthorized = authorizedHubs[relayManager][msg.sender].removalTime == type(uint256).max;
+        bool isRelayOwner = info.owner == msg.sender;
```

Comment:
this method is. called from relayHub, so msg.sender is the hub, and thus its not an owner

Reply:
There is an "OR" relationship for these two checks - `isHubAuthorized OR isRelayOwner`